freqtrade/persistence/models.py

```diff
@@ -403,31 +459,63 @@ def adjust_stop_loss(self, current_price: float, stoploss: float,
             f"Trailing stoploss saved us: "
             f"{float(self.stop_loss) - float(self.initial_stop_loss):.8f}.")
 
+    def is_opening_trade(self, side) -> bool:
+        """
+        Determines if the trade is an opening (long buy or short sell) trade
+        :param side (string): the side (buy/sell) that order happens on
+        """
+        return (side == 'buy' and not self.is_short) or (side == 'sell' and self.is_short)
+
+    def is_closing_trade(self, side) -> bool:
+        """
+        Determines if the trade is an closing (long sell or short buy) trade
+        :param side (string): the side (buy/sell) that order happens on
+        """
+        return (side == 'sell' and not self.is_short) or (side == 'buy' and self.is_short)
+
     def update(self, order: Dict) -> None:
         """
         Updates this entity with amount and actual open/close rates.
         :param order: order retrieved by exchange.fetch_order()
         :return: None
         """
         order_type = order['type']
+
+        if ('leverage' in order and 'borrowed' in order):
+            raise OperationalException(
+                'Pass only one of Leverage or Borrowed to the order in update trade')
+
+        if 'is_short' in order and order['side'] == 'sell':
+            # Only set's is_short on opening trades, ignores non-shorts
+            # TODO-mg: I don't like this, but it might be the only way
+            self.is_short = order['is_short']
+
         # Ignore open and cancelled orders
         if order['status'] == 'open' or safe_value_fallback(order, 'average', 'price') is None:
             return
 
         logger.info('Updating trade (id=%s) ...', self.id)
 
-        if order_type in ('market', 'limit') and order['side'] == 'buy':
+        if order_type in ('market', 'limit') and self.is_opening_trade(order['side']):
             # Update open rate and actual amount
+
             self.open_rate = float(safe_value_fallback(order, 'average', 'price'))
             self.amount = float(safe_value_fallback(order, 'filled', 'amount'))
+            if 'borrowed' in order:
+                self.borrowed = order['borrowed']
+            elif 'leverage' in order:
+                self.leverage = order['leverage']
```

Comment:
the fields used should (must) correspond to the ccxt return object ... and i'm yet to see these fields :(

Reply:
Does this mean that `leverage` and `borrowed` cannot be updated within `trade.update` and that they either must be passed when instantiating a Trade object

```
trade = Trade(
        id=2,
        pair='ETH/BTC',
        stake_amount=0.001,
        open_rate=0.01,
        amount=5,
        is_open=True,
        open_date=ten_minutes_ago,
        fee_open=fee.return_value,
        fee_close=fee.return_value,
        borrowed=90.99181073,
        is_short=True,
        exchange='binance'
    )
 ```
 
 Or must be updated via some other method?
 
 What do you  think would be a good solution?

Reply:
I guess https://github.com/freqtrade/freqtrade/pull/5157#discussion_r660001578 may point to a solution to this (which would enable these keys to be used here) ... but i'm not yet decided on that finally ...

Reply:
Im not exactly sure what the ccxt return object is but here's where leverage is specified for binance and kraken
https://i.imgur.com/CrNjQEi.png